### ISSUE TITLE
girvan_newman's final clusters

### ISSUE THREAD

**Issue by dougcirqueira** [iit-cs579/main#82]
I would like some help on the result returned by this method. Everything else is accordingly with the professor's output, however my final clusters output is like this.

final cluster sizes:
[701, 11, 695, 6, 694, 1, 693, 1, 692, 1, 691, 1, 689, 2, 688, 1, 641, 47, 634, 7, 633, 1, 632, 1, 631, 1, 630, 1, 629, 1, 628, 1, 600, 28, 599, 1, 598, 1, 597, 1, 557, 40, 554, 3, 553, 1, 546, 7, 545, 1, 544, 1, 543, 1, 542, 1, 541, 1, 540, 1, 539, 1, 538, 1, 537, 1, 536, 1, 535, 1, 534, 1, 533, 1, 532, 1, 531, 1, 530, 1, 529, 1, 528, 1, 526, 2, 525, 1, 524, 1, 139, 385, 136, 3, 135, 1, 134, 1, 133, 1, 123, 10, 115, 8, 114, 1, 113, 1, 112, 1, 111, 1, 11, 100, 6, 12, 33, 14, 9, 9, 9, 19, 20, 20, 10, 23, 14, 45, 11, 10, 2, 10, 18, 22, 14, 36, 23, 14, 8, 12, 8, 3, 32, 13, 22, 12, 35, 34, 18, 9, 8, 4, 18, 12, 3, 12, 8, 15, 41, 10, 20, 20, 11, 11, 14, 5, 4, 21, 10, 23, 18, 6, 24, 3, 4, 12, 10, 23, 7, 9, 14, 9, 10, 36, 3, 8, 21, 28, 19, 9, 27, 3, 15, 13, 13, 22, 3, 34, 10, 9, 4, 7, 3, 19, 11, 8, 16, 18, 10, 11, 9, 7, 13, 35, 384, 1, 383, 1, 382, 1, 358, 24, 357, 1, 356, 1, 355, 1, 354, 1, 338, 16, 337, 1, 323, 14, 322, 1, 321, 1, 320, 1, 306, 14, 305, 1, 304, 1, 295, 9, 294, 1, 293, 1, 283, 10, 282, 1, 281, 1, 280, 1, 257, 23, 255, 2, 254, 1, 253, 1, 252, 1, 251, 1, 250, 1, 249, 1, 248, 1, 247, 1, 246, 1, 214, 32, 213, 1, 212, 1, 211, 1, 202, 9, 195, 7, 194, 1, 193, 1, 192, 1, 191, 1, 190, 1, 189, 1, 188, 1, 187, 1, 183, 4, 182, 1, 181, 1, 180, 1, 179, 1, 178, 1, 177, 1, 68, 109, 11, 25, 13, 24, 39, 54, 29, 17, 10, 14, 19, 8, 7, 28, 17, 22, 33, 20, 33, 16, 18, 18, 23, 26, 25, 17, 40, 16, 17, 31, 21, 15, 46, 25, 20, 17, 9, 19, 67, 57, 35, 14, 16, 15, 4, 71, 23, 51, 72, 13, 24, 12, 24, 19, 16, 20, 61, 14, 15, 27, 7, 10, 7, 28, 60, 25, 21, 18, 108, 1, 107, 1, 100, 7, 33, 24, 49, 15, 47, 39, 16, 14, 41, 5, 15, 13, 58, 5, 10, 10, 20, 25, 9, 6, 18, 51, 7, 40, 43, 18, 23, 47, 45, 54, 32, 49, 36, 10, 10, 45, 44, 40, 16, 25, 6, 13, 48, 7, 27, 41, 39, 26, 6, 27, 16, 30, 8, 31, 12, 26, 49, 21, 6, 50, 19, 54, 15, 27, 31, 26, 55, 25, 9, 20, 10, 12, 11, 45, 9, 52, 10, 16, 30, 6, 14, 30, 26, 16, 4, 34, 20, 43, 61, 42, 25, 4, 34, 3, 29, 50, 8, 15, 56, 5, 21, 15, 35, 49, 15, 10, 13, 14, 26, 8, 20, 7, 16, 33, 24, 8, 17, 18, 42, 13, 20, 25, 13, 5, 7, 12, 19, 9, 11, 24, 29, 15, 6, 12, 2, 17, 8, 17, 15, 7, 5, 4, 9, 24, 3, 46, 15, 16, 6, 5, 3, 7, 13, 4, 4, 9, 9, 11, 18, 7, 22, 15, 18, 7, 7, 13, 17, 12, 5, 4, 4, 9, 13, 44, 58, 30, 37, 16, 19, 39, 20, 47, 40, 45, 47, 21, 28, 36, 20, 39, 7, 19, 26, 17, 14, 19, 22, 31, 18, 31, 39, 39, 41, 40, 14, 48, 28, 52, 29, 14, 39, 48, 30, 18, 39, 32, 30, 38, 28]

**Comment by odutfoy** [comment]
That is because you add every cluster to your result in your algorithm
every time you go through your loop. Only the ones between the given size
(50 and 300 I believe)  should be added
Le 2 oct. 2015 23:08, "Douglas Cirqueira" <notifications@github.com> a
écrit :

**Comment by jaydeeppatel1701** [comment]
you are storing all component size. You have to store only cluster whose size is between minsize and maxsize

**Comment by benjaminwalters** [comment]
I'm guessing you started with the code given in lec05. That code has the same problem you're having. I think the problem is the line 
result = [c.nodes() for c in clusters]
It adds the two clusters before continuing the recursion. So it never tests their sizes. I think it should be an empty list instead.

**Comment by dougcirqueira** [comment]
Hi guys! I appreciate all of your answers!
In fact @odutfoy  and @jaydeeppatel1701 , the problem was with the way I was extending my final result list. Instead of returning a list of lists, I was returning a list of with all nodes per cluster. In addition, I started using the code given in lec05, and @benjaminwalters is right about the line "result = [c.nodes() for c in clusters]", which should be an empty list instead. 
I got it! Thank you guys! All your inputs helped me to clarify my problem.
